### PR TITLE
feat: add tagging support

### PR DESCRIPTION
**Pull request by olavloite**
Adds support for request and transaction tags.

Replaces #1286

### CODE REVIEW

**Comment by skuruppu** [comment]
@olavloite I don't think we'd be able to merge this without AVaksman doing `@googlebot I consent`. Were you unable to push commits to his branch directly?

**Review by skuruppu — APPROVED** [review]

**Comment by skuruppu** [comment]
@AVaksman if you're still able to, would you be able to reply with `@googlebot I consent.`?

**Comment by olavloite** [comment]
> @olavloite I don't think we'd be able to merge this without AVaksman doing `@googlebot I consent`. Were you unable to push commits to his branch directly?
> 

I can't push directly to the original branch, as that was in a fork in his account. I'll take a look and see if I can work around it. One alternative is always to just start a brand new branch and copy the code changes in there.

File: src/database.ts [hunk 1]
```diff
@@ -2662,6 +2662,10 @@ class Database extends common.GrpcServiceObject {
     while (true) {
       try {
         const [session, transaction] = await promisify(getWriteSession)();
+        transaction.requestOptions = Object.assign(
```
[the remark this answers is not in the document]
Comment (olavloite) [reply src/database.ts:2665]: No, these batch transactions are read-only transactions and should not support transaction tags.

**Pull request merged**